File: src/main/java/com/technews/common/constant/SpringRepository.java
package com.technews.common.constant;

import lombok.AllArgsConstructor;

@AllArgsConstructor
public enum SpringRepository {
    SPRING_FRAMEWORK("spring-framework"),
    SPRING_BOOT("spring-boot"),
    SPRING_DATA_JPA("spring-data-jpa"),
    SPRING_BATCH("spring-batch"),
    ;

    private String value;

    public String value() {
        return value;
    }
}

File: src/main/kotlin/com/technews/common/constant/SpringRepository.kt
package com.technews.common.constant

enum class SpringRepository(
    val value: String,
) {
    SPRING_FRAMEWORK("spring-framework"),
    SPRING_BOOT("spring-boot"),
    SPRING_DATA_JPA("spring-data-jpa"),
    SPRING_BATCH("spring-batch"),
    ;

    companion object {
        fun list(): List<String> =
            entries.map { it.value }
    }
}
